contracts/escrow/escrow.sol

```diff
@@ -0,0 +1,279 @@
+pragma solidity ^0.5.17;
+
+import "../openzeppelin/SafeMath.sol";
+import "../interfaces/IERC20.sol";
+
+/**
+ *  @title A holding contract for Sovryn Ethereum Pool to accept SOV Token.
+ *  @author Franklin Richards - powerhousefrank@protonmail.com
+ *  @notice You can use this contract for deposit of SOV tokens for some time and withdraw later.
+ */
+contract Escrow {
+	using SafeMath for uint256;
+
+	/* Storage */
+
+	/// @notice The total tokens deposited.
+	/// @dev Used for calculating the reward % share of users related to total deposit.
+	uint256 public totalDeposit;
+	/// @notice The release timestamp for the tokens deposited.
+	uint256 public releaseTime;
+	/// @notice The amount of token we would be accepting as deposit at max.
+	uint256 public depositLimit;
+
+	/// @notice The SOV token contract.
+	IERC20 public SOV;
```

Comment:
Please correct me if I am wrong, as we are using the 0.5.17 compiler, and `immutable` being introduced in 0.6.5, I think it might not be good to change that.

Reply:
oh, sure. You are correct